src/System.IdentityModel.Tokens.Jwt/JwtSecurityTokenHandler.cs

```diff
@@ -558,12 +560,12 @@ private JwtSecurityToken CreateJwtSecurityTokenPrivate(string issuer, string aud
             LogHelper.LogInformation(LogMessages.IDX12722, rawHeader, rawPayload, rawSignature);
 
             if (encryptingCredentials != null)
-                return EncryptToken(new JwtSecurityToken(header, payload, rawHeader, rawPayload, rawSignature), encryptingCredentials);
-            else
-                return new JwtSecurityToken(header, payload, rawHeader, rawPayload, rawSignature);
+                return EncryptToken(new JwtSecurityToken(header, payload, rawHeader, rawPayload, rawSignature), encryptingCredentials, tokenType);
+
+            return new JwtSecurityToken(header, payload, rawHeader, rawPayload, rawSignature);
```

Comment:
Why do we include the token type for encrypted but not for un-encrypted?

Reply:
For un-encrypted the token type is included while creating `Jwtheader`.
Here it is included again so that the 'typ' claim in the outer header also have the token type value. Does that makes sense?

Reply:
makes sense, thanks!